File: backend/src/main/java/ru/fbtw/thestore/backend/data/user/dto/MyUserDto.java
package ru.fbtw.thestore.backend.data.user.dto;

import com.fasterxml.jackson.annotation.JsonView;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.*;
import ru.fbtw.thestore.backend.data.catalog.dto.ProductDto;
import ru.fbtw.thestore.backend.data.geo.dto.CityDto;
import ru.fbtw.thestore.backend.data.order.dto.MyOrderDto;
import jakarta.validation.constraints.Pattern;

import java.io.Serializable;
import java.util.Set;

/**
 * A DTO for the {@link ru.fbtw.thestore.backend.domains.user.MyUser} entity
 */
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Getter
@Setter
public class MyUserDto implements Serializable {

	private Long id;
	@NotBlank(message = "Sex is required")
	private String sex;
	@NotBlank(message = "Username is required")
	private String username;
	@Email(message = "Invalid email address")
	@NotBlank(message = "Email address is required")
	private String userEmail;
	@Pattern(regexp = "^\\+7\\d{10}$",
			message = "Phone number must be in the format +79999999999")
	@NotBlank(message = "Phone number is required")
	private String userPhone;
	private boolean userAnonimus;
	private Long userFirebase;
	private Set<MyOrderDto> orders;
	private Set<ProductDto> basket;
	private Set<ProductDto> favourites;
	private CityDto city;
}

File: mobile_app/the_store_app/lib/entity/user/my_user_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:the_store_app/entity/entity.dart';
part 'my_user_dto.freezed.dart';
part 'my_user_dto.g.dart';


@Freezed(
  fromJson: true,
  toJson: true,
)
class MyUserDto with _$MyUserDto{
  factory MyUserDto({
    String? username,
    String? userEmail,
    String? userPhone,
    String? gender,
    bool? userAnonimus,
    int? userFirebase,
    List<MyOrderDto>? orders,
    List<ProductDto>? basket,
    List<ProductDto>? favourites,
    City? city,
}) = _MyUserDto;

  factory MyUserDto.fromJson(Map<String, dynamic> json) => _$MyUserDtoFromJson(json);
}